### PR TITLE
support fetching ingresses via networking/v1, prune stale metrics, upgrade to use go modules

### PR DESCRIPTION
**Pull request by com6056**
This adds the ability for us to grab the list of ingresses via networking/v1 API. It keeps the default behavior as using extensions/v1beta1 so we don't break existing setups, but we can add `-ingressAPIVersion 'networking/v1'` to support newer versions of Kubernetes.

I also upgraded this repo to use Go modules and added the ability to pass in the Kubernetes context to use to make testing locally easier 🙌 

It also adds pruning of stale metrics for pods that go away 🧹 

```
go run ./... -kubeconfig ~/.kube/config -context REDACTED -loglevel debug -ingressNamespaces REDACTED -labels 'app.kubernetes.io/name=traefik' -namespaces REDACTED -ingressAPIVersion 'networking/v1'
INFO[0000] Starting Prometheus metrics endpoint on :8888
DEBU[0000] Polling
DEBU[0000] discovered host: consul.REDACTED  ns=REDACTED
DEBU[0000] discovered host: alertmanager.REDACTED  ns=REDACTED
DEBU[0000] discovered host: jaeger.REDACTED  ns=REDACTED
DEBU[0000] discovered host: kibana.REDACTED  ns=REDACTED
DEBU[0000] discovered host: prometheus.REDACTED  ns=REDACTED
DEBU[0000] Number of matching pods found in namespace: 1  ns=REDACTED
^CINFO[0002] Received termination signal, shutting down
INFO[0002] Monitor stopping
INFO[0002] Shutdown finished, exiting
```

```
docker build .
[+] Building 56.5s (15/15) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                   0.0s
 => => transferring dockerfile: 37B                                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                                      0.0s
 => => transferring context: 2B                                                                                                                                                        0.0s
 => [internal] load metadata for docker.io/library/alpine:3                                                                                                                            1.2s
 => [internal] load metadata for docker.io/library/golang:1.17                                                                                                                         1.3s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                          0.0s
 => [auth] library/golang:pull token for registry-1.docker.io                                                                                                                          0.0s
 => [stage-1 1/3] FROM docker.io/library/alpine:3@sha256:21a3deaa0d32a8057914f36584b5288d2e5ecc984380bc0118285c70fa8c9300                                                              0.0s
 => [stage-0 1/4] FROM docker.io/library/golang:1.17@sha256:301609ebecc0ec4cd3174294220a4d9c92aab9015b3a2958297d7663aac627a1                                                           0.0s
 => [internal] load build context                                                                                                                                                      0.0s
 => => transferring context: 7.16kB                                                                                                                                                    0.0s
 => CACHED [stage-0 2/4] WORKDIR /src                                                                                                                                                  0.0s
 => [stage-0 3/4] COPY . ./                                                                                                                                                            0.1s
 => [stage-0 4/4] RUN CGO_ENABLED=0 GOOS=linux go build -o app                                                                                                                        54.8s
 => CACHED [stage-1 2/3] RUN apk --no-cache add ca-certificates                                                                                                                        0.0s
 => CACHED [stage-1 3/3] COPY --from=0 /src/app /app                                                                                                                                   0.0s
 => exporting to image                                                                                                                                                                 0.0s
 => => exporting layers                                                                                                                                                                0.0s
 => => writing image sha256:8634a65c72dd8ff2085b8230f2a19a94706f4c4fae2a3176b6db3994f8a22f37                                                                                           0.0s
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -15,8 +15,8 @@ import (
 	"time"
 
 	"github.com/muxinc/certificate-expiry-monitor/monitor"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
```
Comment (dlmather) [inline main.go:18]: this should be in the import section below

go native

package local

external packages

**Review by dlmather — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -37,6 +38,7 @@ var (
 	logFormat          = flag.String("logformat", "text", "Log format (text or json)")
 	metricsPort        = flag.Int("metricsPort", 8888, "TCP port that the Prometheus metrics listener should use")
 	insecureSkipVerify = flag.Bool("insecure", true, "If true, then the InsecureSkipVerify option will be used with the TLS connection, and the remote certificate and hostname will be trusted without verification")
+	ingressAPIVersion  = flag.String("ingressAPIVersion", "extensions/v1beta1", "Version of the Ingress API to use, can be either `extensions/v1beta1` or `networking/v1`")
```
Comment (sushraju) [inline main.go:41]: Can we use the same pattern we use in other places?  That is, if we pass ingressApiVersion as "networking/v1" then the new Ingress API can be used?  Thoughts?

Comment (com6056) [reply main.go:41]: Don't think we can since we have to explicitly add a case for it and add the clientset to use so it can't really be dynamic

**Review by sushraju — APPROVED** [review]
🔥

**Pull request merged**